rust-toolchain.toml

```diff
@@ -1,5 +1,10 @@
 [toolchain]
-# Please update rust-version in **/Cargo.toml files when changing channel:
-channel = "1.56.0"
+# This specified the version we use to build.
+channel = "1.57.0"
+# In order to provide provide backward compatibility, we define minimum version.
+# Current minimum version is "1.56.0", please change rust-version in `**/Cargo.toml`.
+# The minimum supported version is important for outside packages, importing packages
+# from `near/nearcore`. That means we are limited to features available in `1.56.0`.
+# However, we can still benifit from compiler optimizations, improved `cargofmt`, `clippy`, etc.
```

Comment:
I feel that I would not be able to follow this comment unless I bring some prior context. Also, the comment seems to be misplaced.

Reply:
@frol Yes, I agree. I'll rewrite this section. I think, I should keep this PR changes to minimum, and the doc in a separate PR. To make bumping version non blocking, and independent from good documentation.

Reply:
@frol Can you take a another look. I rewrote the documentation based on the feedback. It should be easier to understand without prior context.